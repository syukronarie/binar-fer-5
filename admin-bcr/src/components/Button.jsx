import React from "react";

const Button = ({ children, ...rest }) => {
	return <button {...rest}>{children}</button>;
};

export default Button;
