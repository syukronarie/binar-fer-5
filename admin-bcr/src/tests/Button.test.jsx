import { render, screen } from "@testing-library/react";
import Button from "../components/Button";

describe("Button Test", () => {
	it("should be match with the snapshot", () => {
		render(<Button type="submit">Submit</Button>);
		expect(screen.getByText(/submit/i)).toMatchSnapshot();
	});
	it("should render button correctly without error", () => {
		render(<Button>Submit</Button>);
		expect(screen.getByText(/submit/i)).toBeInTheDocument();
	});
});
