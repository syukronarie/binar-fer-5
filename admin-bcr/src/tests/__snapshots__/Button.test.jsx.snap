// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test should be match with the snapshot 1`] = `
<button
  type="submit"
>
  Submit
</button>
`;
